src/picasso-definition/components/mini-chart/background-rect.js

```diff
@@ -0,0 +1,26 @@
+import KEYS from '../../../constants/keys';
+import NUMBERS from '../../../constants/numbers';
+
+export default function createMiniChartBackgroundRect({ models, flags }) {
+  const ratio = NUMBERS.MINI_CHART.RATIO;
+  const padding = NUMBERS.MINI_CHART.PADDING; // Padding from the bottom right corner
+  let width;
+  let height;
+  const { chartModel, layoutService } = models;
+  const viewHandler = chartModel.query.getViewHandler();
+  return {
+    key: KEYS.COMPONENT.MINI_CHART_BACKGROUND,
+    type: 'custom-rect',
+    style: { borderColor: 'black', background: 'rgba(255, 255, 255, 0.7)' },
+    rect: {
+      x: () => width * (1 - ratio) - padding,
+      y: () => height * (1 - ratio) - padding,
+      width: () => width * ratio,
+      height: () => height * ratio,
+    },
+    show: () => viewHandler.getMeta().scale < 1 && layoutService.meta.isBigData && flags.isEnabled('DATA_BINNING'),
```

Comment:
If the flag is not enabled, you can just return false earlier at line 5, instead of creating the component at all. Should be same for `layoutService.meta.isBigData` and everything that can be resolved during creation time. `show` should only be used for stuff that can change during "runtime" of the chart, for example zooming/panning.

Reply:
Looking more, you can probably return `false` in mini-chart/index.js for everything that can be resolved at that point already (I assume the 3 components use the same show condition). Then only leave stuff in the show function that can change during zoom/pan etc, if anything.

Reply:
good point!